python/cudf/cudf/core/frame.py

```diff
@@ -139,6 +153,29 @@ def _from_data(
         Frame.__init__(obj, data, index)
         return obj
 
+    @classmethod
+    def _from_maybe_indexed_columns(
+        cls,
+        columns: List[ColumnBase],
+        column_names: List[str],
+        index_names: Optional[List[str]] = None,
+    ):
+        index = None
+        if index_names is not None:
+            index_ids = list(range(len(index_names)))
```

Comment:
This range can be used directly in the `zip` call.
```suggestion
            index_ids = range(len(index_names))
```